StartBonusActivation/handler.ts

```diff
@@ -196,23 +232,46 @@ const getLastValidDSU = (
 const createBonusActivation = (
   bonusActivationModel: BonusActivationModel,
   fiscalCode: FiscalCode,
-  dsu: Dsu
-): TaskEither<IResponseErrorInternal, BonusActivation> =>
-  fromQueryEither(() => {
-    // TODO: generate bonus code with provided algorithm
-    const bonusCode = "any code" as BonusCode & NonEmptyString;
-    const bonusActivation: NewBonusActivation = {
-      applicantFiscalCode: fiscalCode,
-      createdAt: new Date(),
-      dsuRequest: dsu,
-      id: bonusCode,
-      kind: "INewBonusActivation",
-      status: BonusActivationStatusEnum.PROCESSING
-    };
-    return bonusActivationModel.create(bonusActivation, fiscalCode);
-  }).mapLeft(err =>
-    ResponseErrorInternal(`Error creating BonusActivation: ${err.message}`)
+  dsu: Dsu,
+  maxAttempts: number = 5
+): TaskEither<IResponseErrorInternal, BonusActivation> => {
+  const shouldRetry = (err: QueryError) => {
+    // CosmosDB conflict: primary key violation
+    return err.code === 409;
+  };
+
+  const retriableBonusActivationTask = tryCatch(
```

Comment:
ok by now but try to avoid closures / nested methods 
(`retriableBonusActivationTask` should be at top level then passed as a parameter to `createBonusActivation`, shouldRetry can be at top level, ...)